examples/gallery/lines/line_segment_ends.py

```diff
@@ -0,0 +1,72 @@
+"""
+Line segment caps and joints
+============================
+PyGMT offers different appearance of line segment caps and joints. The desired
+appearance can be set via the GMT default parameters :gmt-term:`PS_LINE_CAP`
+(``"butt"``, ``"round"``, or ``"square"`` [Default]) as well as :gmt-term:`PS_LINE_JOIN`
+(``"bevel"``, ``"round"``, and ``"miter"`` [Default]) and :gmt-term:`PS_MITER_LIMIT`
+(limit on the angle at the mitered joint below which a bevel is applied).
+"""
+
+# %%
+import numpy as np
+import pygmt
+
+# Set up dictionary for colors
+dict_col = {
+    "round": "green4",
+    "square": "steelblue",
+    "butt": "orange",
+    "miter": "steelblue",
+    "bevel": "orange",
+}
+
+# Create new Figure instance
+fig = pygmt.Figure()
+
+# -----------------------------------------------------------------------------
+# Top: PS_LINE_CAP
+
+# Create sample data
+x = np.array([30, 170])
+y = np.array([70, 70])
+
+fig.basemap(region=[0, 250, 0, 100], projection="x1p", frame="rltb")
+
+for line_cap in ["butt", "round", "square"]:
+    # Change GMT default locally using a context manager
+    # The change applies only to the code under the with statement
+    with pygmt.config(PS_LINE_CAP=line_cap):
+        # Plot a dashed line
+        fig.plot(x=x, y=y, pen="10p," + dict_col[line_cap] + ",20_20:0")
```

Comment:
Make it more readable:
```suggestion
        # Plot a dashed line
        color = dict_col[line_cap]
        fig.plot(x=x, y=y, pen=f"10p,{color},20_20:0")
```